cli/azd/internal/telemetry/telemetry_test.go

```diff
@@ -7,29 +7,35 @@ import (
 	"testing"
 
 	"github.com/azure/azure-dev/cli/azd/internal"
+	appinsightsexporter "github.com/azure/azure-dev/cli/azd/internal/telemetry/appinsights-exporter"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 func TestGetTelemetrySystem(t *testing.T) {
+	devEndpointConfig, err := appinsightsexporter.NewEndpointConfig(devConnectionString)
+	require.NoError(t, err)
+	prodEndpointConfig, err := appinsightsexporter.NewEndpointConfig(prodConnectionString)
+	require.NoError(t, err)
+
 	type args struct {
 		version                     string
 		disableTelemetryEnvVarValue string
 	}
 	tests := []struct {
-		name       string
-		args       args
-		expectNil  bool
-		expectIKey string
+		name         string
+		args         args
+		expectNil    bool
+		expectConfig appinsightsexporter.EndpointConfig
 	}{
-		{"DevVersion", args{"0.0.0-dev.0 (commit 0000000000000000000000000000000000000000)", "unset"}, false, devInstrumentationKey},
-		{"DevVersionTelemetryEnabled", args{"0.0.0-dev.0 (commit 0000000000000000000000000000000000000000)", "yes"}, false, devInstrumentationKey},
-		{"DevVersionTelemetryDisabled", args{"0.0.0-dev.0 (commit 0000000000000000000000000000000000000000)", "no"}, true, devInstrumentationKey},
+		{"DevVersion", args{"0.0.0-dev.0 (commit 0000000000000000000000000000000000000000)", "unset"}, false, devEndpointConfig},
+		{"DevVersionTelemetryEnabled", args{"0.0.0-dev.0 (commit 0000000000000000000000000000000000000000)", "yes"}, false, devEndpointConfig},
+		{"DevVersionTelemetryDisabled", args{"0.0.0-dev.0 (commit 0000000000000000000000000000000000000000)", "no"}, true, devEndpointConfig},
 
 		// Currently, prod version should always be disabled.
-		{"ProdVersion", args{"1.0.0 (commit 13ec2b11aa755b11640fa16b8664cb8741d5d300)", "no"}, true, prodInstrumentationKey},
-		{"ProdVersion", args{"1.0.0 (commit 13ec2b11aa755b11640fa16b8664cb8741d5d300)", "unset"}, true, prodInstrumentationKey},
-		{"ProdVersion", args{"1.0.0 (commit 13ec2b11aa755b11640fa16b8664cb8741d5d300)", "yes"}, true, prodInstrumentationKey},
+		{"ProdVersion", args{"1.0.0 (commit 13ec2b11aa755b11640fa16b8664cb8741d5d300)", "no"}, true, prodEndpointConfig},
```

Comment:
Is the comment on L35 still valid?  I think it should be removed (and that's why the `true`'s change to `false`'s here in the test, because we expect to get a telemetry system in prod when it is not explicitly turned off?

Reply:
Lost in merge -- will fix